server/services/v1/search_indexer.go

```diff
@@ -239,3 +242,40 @@ func UnpackAndSetMD(doc map[string]interface{}, tableData *internal.TableData) {
 		tableData.UpdatedAt = internal.CreateNewTimestamp(int64(v.(float64)))
 	}
 }
+
+func FlattenObjects(input map[string]any) map[string]any {
+	resp := make(map[string]any)
+	for k, v := range input {
+		switch vType := v.(type) {
+		case map[string]any:
+			flattened := FlattenObjects(vType)
+			for nestedK, nestedV := range flattened {
+				resp[k+schema.ObjFlattenDelimiter+nestedK] = nestedV
+			}
+		default:
+			resp[k] = v
+		}
+	}
+
+	return resp
+}
+
+func UnFlattenObjects(flat map[string]any) map[string]any {
```

Comment:
I think we can do it more efficient, and avoid dependency on `mergo`:
```
result := make(map[string]any)

for k, v := range flat {
   keys := strings.Split(k, schema.ObjFlattenDelimiter)
   m := result // top level map always exists
   for i := 1; i < len(keys); i++ { // create submaps if necessary
      if _, ok := m[keys[i-1]; !ok {
         m[keys[i-1] = make(map[string]any)
      }
      m = map[keys[i-1]]
   }
   // here m contains last submap
   m[keys[len(keys)-1] = v
}
```

Reply:
let me change it.